README.rst

```diff
@@ -730,6 +730,49 @@ reset to empty state to avoid potential conflicts.
 Set this to ``False`` if every organization has its dedicated management
 tunnel with a dedicated address space that is reachable by the OpenWISP server.
 
+``OPENWISP_CONTROLLER_CUSTOM_OS_MAPPING``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+----------+
+| **type**:    | ``dict`` |
++--------------+----------+
+| **default**: | ``{}``   |
++--------------+----------+
+
+A ``dict`` mapping OS identifier of your custom firmware to the OpenWrt
+configuration version. The value of this setting is used to decide
+syntax of device's configuration.
```

Comment:
I would rewrite it along these lines:

OpenWISP Controller can figure out whether it should use the new OpenWrt syntax for DSA interfaces (Distributed Switch Architecture) introduced in OpenWrt 21 or not by reading the ``os`` field of the ``Device`` object.
However, if the firmware you are using has a custom firmware identifier, the system will not be able to figure out whether the new syntax should be used or not and it will default to ``OPENWISP_CONTROLLER_DSA_DEFAULT_FALLBACK``.

If you want to make sure the system can parse your custom firmware identifier properly, you can follow the example below.

For the sake of the example, the OS identifier `MyCustomFirmware 2.0` corresponds to OpenWrt 19.07, while `MyCustomFirmware 2.1` corresponds to OpenWrt 21.02. Configuring this setting as indicated below will allow OpenWISP to supply the right syntax automatically.